src/lib/fpti.js

```diff
@@ -0,0 +1,134 @@
+/* @flow */
+import { getClientID, getMerchantID, getPartnerAttributionID } from '@paypal/sdk-client/src';
+
+import { getDeviceInfo } from './get-device-info';
+
+const sendBeacon = (src, data) => {
+    if (!src || !data) {
+        return;
+    }
+
+    let query = Object.keys(data).map(key => {
+        return `${ encodeURIComponent(key) }=${ encodeURIComponent(data[key]) }`;
+    });
+
+    query = query.join('&');
+
+    const beaconImage = new window.Image();
+    beaconImage.src = `${ src }?${ query }`;
+};
+
+// removes empty strings, `undefined`, `null`, and `NaN` from fpti event
+const filterFalsyValues = source => {
+    Object.keys(source).forEach(key => {
+        const val = source[key];
+
+        if (val === '' || val === undefined || val === null || Number.isNaN(val)) {
+            delete source[key];
+        }
+    });
+
+    return source;
+};
+
+const resolveTrackingData = (config, data) => {
+    const deviceInfo = getDeviceInfo();
+
+    return {
+        e: 'im',
+        comp: 'ppshoppingsdk',
+        page: `ppshopping:${ data.eventName }`,
+        t: new Date().getTime(),
+        g: new Date().getTimezoneOffset(),
+        ...deviceInfo,
+        ...config,
+        ...data
+    };
+};
+
+const resolveTrackingVariables = (data) => ({
+    // Device height
+    dh: data.screenHeight,
+
+    // Device width
+    dw: data.screenWidth,
+
+    // Browser height
+    bh: data.browserHeight,
+
+    // Browser width
+    bw: data.browserWidth,
+
+    // Color depth
+    cd: data.colorDepth,
+
+    // Screen height
+    sh: data.screenHeight,
+
+    // Screen width
+    sw: data.screenWidth,
+
+    // Device type
+    dvis: data.deviceType,
+
+    // Browser type
+    btyp: data.browserType,
+
+    // Rosetta language
+    rosetta_language: data.rosettaLanguage,
+
+    // Page domain & path
+    ru: data.location,
+
+    // Identification confidence score
+    confidence_score: data.confidenceScore,
+
+    // Identification type returned by VPNS
+    identifier_used: data.identifierUsed,
+
+    // Unverified encrypted customer account number
+    unverified_cust_id: data.userEAN,
```

Comment:
These vpns values won't be present until we integrate identity on the client, right?

Reply:
Yes.